scalding-parquet-scrooge/src/main/scala/com/twitter/scalding/parquet/scrooge/PartitionedParquetScroogeSource.scala

```diff
@@ -0,0 +1,46 @@
+package com.twitter.scalding.parquet.scrooge
+
+import _root_.cascading.scheme.Scheme
+import _root_.cascading.tuple.Fields
+import com.twitter.scalding._
+import com.twitter.scalding.parquet.thrift.ParquetThriftBase
+import com.twitter.scalding.typed.{ PartitionSchemed, PartitionUtil }
+import com.twitter.scrooge.ThriftStruct
+
+/**
+ * Scalding source to read or write partitioned Parquet scrooge data.
+ *
+ * For writing it expects a pair of `(P, T)`, where `P` is the data used for partitioning and
+ * `T` is the scrooge object. Below is an example.
+ * {{{
+ * val data = MyScroogeObject()
+ * IterablePipe(data, flowDef, mode)
+ *  .write(PartitionedParquetScroogeSource[(String, String), MyScroogeObject](path, "%s/%s"))
+ * }}}
+ *
+ * For reading it produces a pair `(P, T)` where `P` is the partition data, `T` is the corresponding
+ * scrooge object. Below is an example.
+ * {{{
+ * val in: TypedPipe[(String, String), MyScroogeObject] =
+ * TypedPipe.from( PartitionedParquetScroogeSource[(String, String), MyScroogeObject](path, "%s/%s") )
+ * }}}
```

Comment:
I can look into this. Not super familiar with Scala macros so this can be a learning opportunity :-). Currently we do check this (at runtime though) - in [TemplatePartition](https://github.com/twitter/scalding/blob/develop/scalding-core/src/main/scala/com/twitter/scalding/typed/TemplatePartition.scala#L28)
```
case class TemplatePartition(partitionFields: Fields, template: String) extends Partition {
  assert(
    partitionFields.size == "%s".r.findAllIn(template).length,
    "Number of partition fields %s does not correspond to template (%s)".format(partitionFields, template))
...
}
```